### DISCUSSION TITLE
Apply texture over text

### DISCUSSION THREAD

**Discussion by LucianPuca** [ImageMagick/ImageMagick#3947] [Help]
Hi guys, wondering if you can help.

I'm looking to apply a gold texture over different jpeg images, they will all have black text with white background.

As an example, I want this texture: https://ibb.co/w4FKjH7 applied over this text image: https://ibb.co/zFBGrLG

After it's applied, I want the newly generated image to be overlayed on top of new image with coloured background, making the white background transparent so only the gold textured text is visible.

So the final result would look something like this: https://ibb.co/cDktDT5 

I hope my rant makes sense, any help is appreciated!

Thank you in advance.

IM on Windows, latest version

**Comment by snibgo** [comment]
Composite the background image over the gold image, using a mask that is black letters on a white backgound.
```
magick gold.png backgrnd.png ( -pointsize 100 label:Gold ) out.png
```

**Reply by LucianPuca** [reply]
Thanks, it doesn't seem to get the result I need.

For the first step (applying the gold texture over the text) I can run something like this:

`convert goldTexture.jpg ( Text.jpg -channel a -negate +channel ) ( +clone -alpha extract -negate +write mpr:alpha +delete ) -compose plus -composite mpr:alpha -alpha off -compose over -compose copy_opacity -composite OverlayTexture.jpg`
It gives me this result: https://ibb.co/TkMGwm6

However, both the gold texture and the text images need to have similar sizes, otherwise it ends up like this: https://ibb.co/wRwbqwT

Not sure if there's a way to keep the output image the same size as the original one (the text image).

**Comment by snibgo** [comment]
That command is confusing. You have:
```
( Text.jpg -channel a -negate +channel )
```
A jpg file has no transparency. It is fully opaque. Negating the alpha channel would make it fully transparent.

```
-compose over -compose copy_opacity -composite
```
"-compose over" is a setting that does nothing at all, because you immediately change the setting with "-compose copy_opacity".

> It gives me this result:
Does it?

You can crop the background and gold images to the size of the text image.

**Reply by LucianPuca** [reply]
Thank you, as you can probably see, I don't quite know what I'm doing, hence me asking for help :) 

I'm still stuck at the part where I overlap the text with gold texture on top of another image, with transparency so only the gold text is visible.

**Comment by fmw42** [comment]
I often find the need to reset the -compose to over after doing a mathematical compose and prior to doing the -compose copy opacity.  The mathematical compose or other compose methods are often left around and change the way -compose copy_opacity works when adding transparency. So it is actually sometimes needed to get correct behavior. At least that was my experience in IM 6.

**Comment by snibgo** [comment]
That would be a bug.
